python/lsst/ts/MTAlignment/alignmentModel.py

```diff
@@ -0,0 +1,507 @@
+        """
+
+        msg = "!SET_WORKING_FRAME:" + workingframe
+        data = await self.send_msg(msg)
+        return data
+
+    async def new_station(self):
+        """
+        A new station is added and made to be the active instrument.
+        """
+
+        msg = "!NEW_STATION"
+        data = await self.send_msg(msg)
+        return data
+
+    async def save_sa_jobfile(self, filepath):
+        """
+        Save a jobfile
+
+        filepath: String
+        """
+
+        msg = "!SAVE_SA_JOBFILE:" + filepath
+        data = await self.send_msg(msg)
+        return data
+
+    async def set_station_lock(self, val):
+        """
+        Prevents SA from automatically jumping stations when it detects that the tracker has drifted.
+
+        val: Boolean
+        """
+        if val:
+            msg = "!SET_STATION_LOCK:1"
+        else:
+            msg = "!SET_STATION_LOCK:0"
+        data = await self.send_msg(msg)
+        return data
+
+    async def reset_t2sa(self):
+        """
+        Reboots the T2SA and SA components
+        """
+
+        msg = "!RESET_T2SA"
+        data = await self.send_msg(msg)
+        return data
+
+    async def halt(self):
+        """
+        Commands T2SA to halt any measurement plans currently executing,
+        and return to ready state
+        """
+
+        msg = "!HALT"
+        data = await self.send_msg(msg)
+        return data
+
+    async def publish_telescope_position(self, telalt, telaz, camrot):
+        """
+        Command in which TCS informs T2SA of the telescope’s current
+        position and camera rotation angle, to be used ahead of a
+        measurement cmd.
+        TODO: we need to add dome position to this for calscreen alignment
+
+        telalt: float
+        telaz: float
+        camrot: float
+        """
+
+        msg = f"PUBLISH_ALT_AZ_ROT:{telalt};{telaz};{camrot}"
+        data = await self.send_msg(msg)
+        return data
+
+    async def set_num_samples(self, numsamples):
+        """
+        This is the number of times the tracker samples each point
+        in order to get one (averaged) measurement
+
+        numsamples: int
+        """
+
+        msg = f"SET_NUM_SAMPLES:{numsamples}"
+        data = await self.send_msg(msg)
+        return data
+
+    async def set_num_iterations(self, numiters):
+        """
+        This is the number of times we repeat an auto-measurement
+        of a point group
+
+        numiters: int
+        """
+
+        msg = f"SET_NUM_ITERATIONS:{numiters}"
+        data = await self.send_msg(msg)
+        return data
+
+    async def increment_measured_index(self, idx):
+        """
+        Increment to measured point group index by idx
+
+        idx: int
+        """
+
+        msg = f"INC_MEAS_INDEX:{idx}"
+        data = await self.send_msg(msg)
+        return data
+
+    async def disconnect(self):
+        self.writer.close()
+        self.connected = False
+
+    def msgformat(self, st):
```

Comment:
I think you only used this method once but is duplicating the logic in line in more than one place. Maybe use it broadly.